CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 if(BUILD_TESTING)
     # tests
+    include(AddGoogleTest)
```

Comment:
I believe there was some reason why we would prefer to add this per directory instead of adding it here on the cmake root.
However, I don't remember why it was. If it works fine, this is good.  I will try to find more info about it.

Reply:
Sure. Feel free to raise an issue on this if you find more information that shows this is not desired.

Reply:
I'm thinking 

> I believe there was some reason why we would prefer to add this per directory instead of adding it here on the cmake root.
> However, I don't remember why it was. If it works fine, this is good. I will try to find more info about it.

+1, I'm thinking about the same thing. In CI, there are some jobs validating without ut and some with ut. So we cannot assume that every sdk depends on GoogleTest. I don't feel it's right to put it here.

Reply:
I think we have reached an agreement to use GTest unless exempted/approved by the Architecture board for a specific reason.
Also, this line cannot be called twice so there are only two solutions available:
1. Put it in the root folder.
2. Use some flags to skip calling this in subsequent subfolders.
I don't know how to do 2 and I think whoever applies for the exemption should be responsible to change 1 to 2, but for now, it is good to put it in the root folder's CMake file.

Reply:
ok, fine